src/test/java/com/scottlogic/filter/FilterANDTest.java

```diff
@@ -0,0 +1,197 @@
+        }
+    }
+
+    @Nested
+    @DisplayName("Content keyword = 'li' AND ...")
+    class ContentKeywordAND {
+
+        FilterContentKeyword filterContentKeyword = new FilterContentKeyword("li");
+
+        @Test
+        @DisplayName("author name = 'Steve'")
+        void filterKeywordIs_AND_authorSteve() {
+            List<UserPost> postsToFilter = UserPostFilterData.getUnfilteredData();
+            List<UserPost> expectedFilteredList = UserPostFilterData.getFilteredKeywordLi_AND_authorSteve();
+            FilterAND filterAND = new FilterAND(filterContentKeyword, new FilterAuthor("Steve"));
+            Assertions.assertEquals(expectedFilteredList, filterAND.filter(postsToFilter));
+        }
+
+        @Test
+        @DisplayName("between dates = '2016-01-01T01:01:01Z 2021-01-01T01:01:01Z'")
+        void filterKeywordIs_AND_betweenDates() {
+            List<UserPost> postsToFilter = UserPostFilterData.getUnfilteredData();
+            List<UserPost> expectedFilteredList = UserPostFilterData.getFilteredKeywordLi_AND_betweenDates();
+            FilterAND filterAND = new FilterAND(filterContentKeyword,
+                    new FilterBetweenDates("2016-01-01T01:01:01Z 2021-01-01T01:01:01Z"));
+            Assertions.assertEquals(expectedFilteredList, filterAND.filter(postsToFilter));
+        }
+
+        @Test
+        @DisplayName("content keyword = 'funky'")
+        void filterKeywordIs_AND_keywordFunky() {
+            List<UserPost> postsToFilter = UserPostFilterData.getUnfilteredData();
+            List<UserPost> expectedFilteredList = UserPostFilterData.getFilteredKeywordLi_AND_keywordFunky();
+            FilterAND filterAND = new FilterAND(filterContentKeyword, new FilterContentKeyword("funky"));
+            Assertions.assertEquals(expectedFilteredList, filterAND.filter(postsToFilter));
+        }
+
+        @Test
+        @DisplayName("liked posts")
+        void filterKeywordIs_AND_likedPosts() {
+            List<UserPost> postsToFilter = UserPostFilterData.getUnfilteredData();
+            List<UserPost> expectedFilteredList = UserPostFilterData.getFilteredKeywordLi_AND_likedPosts();
+            FilterAND filterAND = new FilterAND(filterContentKeyword, new FilterLikedPosts());
+            Assertions.assertEquals(expectedFilteredList, filterAND.filter(postsToFilter));
+        }
+    }
+
+    @Test
+    @DisplayName("Empty input list")
+    void setFilterAND_emptyInput() {
+        List<UserPost> postsToFilter = Arrays.asList();
+        FilterAND filterAND = new FilterAND(new FilterAuthor("Matt"), new FilterContentKeyword("funk"));
+        Assertions.assertEquals(Arrays.asList(), filterAND.filter(postsToFilter));
+    }
+
+    @Test
+    @DisplayName("Null input list")
+    void setFilterAND_nullInput() {
+        FilterAND filterAND = new FilterAND(new FilterAuthor("Matt"), new FilterContentKeyword("funky"));
+        Assertions.assertEquals(Arrays.asList(), filterAND.filter(null));
+    }
```

Comment:
Nice edge cases. What happens when we construct and and filter like this?

`FilterAND filterAND = new FilterAND(null, new FilterContentKeyword("funky"));`.

Is this a relevant edge case?

Reply:
I'd say it is definitely a relevant edge case - have tested on both OR/AND filters and it works appropriately!